inspirehep/modules/disambiguation/api.py

```diff
@@ -37,38 +38,50 @@
 from inspirehep.modules.disambiguation.utils import open_file_in_folder
 
 
-def save_signatures_and_clusters():
-    """Save signatures and starting clusters to disk.
+def save_curated_signatures_and_input_clusters():
+    """Save curated signatures and input clusters to disk.
 
-    Saves two files to disk called (by default) ``clusters.json`` and ``signatures.jl``.
-    The former contains a dictionary that represents the starting clusters that are used
-    by ``BEARD`` as ground truth, while the latter contains one line per curated signature
-    present in INSPIRE.
+    Saves two files to disk called (by default) ``input_clusters.jsonl`` and
+    ``curated_signatures.jsonl``. The former contains one line per each cluster
+    initially present in INSPIRE, while the latter contains one line per each
+    curated signature that will be used as ground truth by ``BEARD``.
     """
-    clusters = defaultdict(list)
+    signatures_with_author = defaultdict(list)
+    signatures_without_author = []
 
-    with open_file_in_folder(current_app.config['DISAMBIGUATION_SIGNATURES_PATH'], 'w') as fd:
+    with open_file_in_folder(current_app.config['DISAMBIGUATION_CURATED_SIGNATURES_PATH'], 'w') as fd:
         for signature in get_all_curated_signatures():
             if signature.get('author_id'):
-                clusters[signature['author_id']].append(signature['signature_uuid'])
+                signatures_with_author[signature['author_id']].append(signature['signature_uuid'])
                 fd.write(json.dumps(signature) + '\n')
-
-    with open_file_in_folder(current_app.config['DISAMBIGUATION_CLUSTERS_PATH'], 'w') as fd:
-        json.dump(clusters, fd)
+            else:
+                signatures_without_author.append(signature['signature_uuid'])
+
+    with open_file_in_folder(current_app.config['DISAMBIGUATION_INPUT_CLUSTERS_PATH'], 'w') as fd:
+        for cluster_id, (author_id, signature_uuids) in enumerate(six.iteritems(signatures_with_author)):
+            fd.write(json.dumps({
+                'author_id': author_id,
+                'cluster_id': cluster_id,
+                'signature_uuids': signature_uuids,
+            }) + '\n')
+        for cluster_id, signature_uuid in enumerate(signatures_without_author, cluster_id + 1):
```

Comment:
...because I agree with you that it's unpleasant that I leverage the fact that `cluster_id` escapes the scope of the `for` loop, and I dislike that I'm using the not-so-well-known two argument form of `enumerate` !